src/main/java/org/ecocean/MetricsBot.java

```diff
@@ -256,6 +272,31 @@ public static void refreshMetrics(String context) {
         else if(contributorsLabels.endsWith(",")) {contributorsLabels="\""+contributorsLabels.substring(0,(contributorsLabels.length()-1))+"\"";}
         csvLines.add(buildGauge("SELECT count(this) FROM org.ecocean.User WHERE username == null", "wildbook_datacontributors_total","Number of public data contributors",context,contributorsLabels));
         
+        //WBIA Metrics pull
+        String metricsURL = IA.getProperty(context, "IBEISIARestUrlAddImages");
+        if(metricsURL!=null) {
+          metricsURL=metricsURL.replaceAll("/api/image/json/","")+"/metrics";
+          String wbiaMetrics=httpGetRemoteText(metricsURL);
+
+          //WBIA turnaround time all task types
+          String regexTT="wbia_turnaround_seconds\\{endpoint=\"\\*\".*\\} \\d*\\.\\d*";
+          if(getWBIAPrometheusClientValue(wbiaMetrics, regexTT)!=null) {
+            csvLines.add("wildbook_wbia_turnaroundtime"+","+getWBIAPrometheusClientValue(wbiaMetrics, regexTT)+","+"gauge"+","+"WBIA job queue turnaround time");
+          }
+          
+          //WBIA turnaround time detection (lightnet) tasks
+          String regexTTdetect="wbia_turnaround_seconds\\{endpoint=\"/api/engine/detect/cnn/lightnet/\".*\\} \\d*\\.\\d*";
+          if(getWBIAPrometheusClientValue(wbiaMetrics, regexTTdetect)!=null) {
+            csvLines.add("wildbook_wbia_turnaroundtime_detection"+","+getWBIAPrometheusClientValue(wbiaMetrics, regexTTdetect)+","+"gauge"+","+"WBIA job queue turnaround time for detection tasks");
+          }
+          
+          //WBIA turnaround time ID (graph) tasks
```

Comment:
for review purposes, i am just going to assume these json regexs do what i am guessing they should do.  :sweat_smile:

Reply:
I tested them in regex101.com and they work in Java too.  :)